### DISCUSSION TITLE
@itwin/core-frontend 3.0.2 mapLayer  bingMaps

### DISCUSSION THREAD

**Discussion by PaoloPetra** [iTwin/itwinjs-core#3490] [Q&A]
Hello everyone
are struggling with a migration of our app from 2.19 to 3.0.2
I am having difficulty with viewing maplayers
I have become aware that to see the maps in this new version you need to pass keys created respectively by bingmaps and by https://cesium.com/
so I'm using the IModelAppOptions interface to pass to the
IModelApp.startup (opt1)

the first question is this BingMaps of (mapLayerOptions) has two values ​​to pass a "key" and a "value" what do they represent respectively?
when I created the key on bing I only have the key and the "application name" field

const opt1: IModelAppOptions = {
   mapLayerOptions: {
        BingMaps: {key: bingMapKey, value: bingMapKey}
      },
      tileAdmin: {cesiumIonKey: "xxxxxxxxxxxxxxxxxx"
     }
 }

consider that inside bingMapKey there is the bingmaps key and not knowing who to attribute it to, I pass it both to the key and to the value

terrain also gives me error even if I have registered and pass the key inside opt1
  
"Terrain is unavailable
The application must be configured with a valid Cesium ION access key to obtain 3d terrain data. "

on the contrary if I pass the data in the view in the mapLayerOptions parameter through "MapLayerOptions" I can see the map.

I am attaching the error of the calls that it seems that I don't have the authorization

![bing_view1](https://user-images.githubusercontent.com/40106886/162189828-85e7b876-ab87-413d-9397-ca450bcdc6b3.png)

![bing_view2](https://user-images.githubusercontent.com/40106886/162189870-70b908b0-34cf-4718-a403-572f71d9fd94.png)

thanks again for your invaluable help

**Comment by pmconne** [comment]
@mdastous-bentley

**Reply by mdastous-bentley** [reply]
Hi Paolo,

BingMaps key can be defined using following structure:
`{BingMaps: { key: "key", value: "YOUR_BING_KEY"}}`
the 'key' value, is in fact parameter name that be inserted in the final request. 
`tileAdmin` option seems to be fine. 

Are you sing the web-viewer-react package, as described in the [Getting Started](https://www.itwinjs.org/getting-started/) page?

Thanks,

Michel

**Reply by pmconne** [reply]
@mdastous-bentley can we document this in MapLayerKey? For example, MapBox wants key="access_token" instead of "key".

**Reply by PaoloPetra** [reply]
hello @mdastous-bentley thanks for the answer, actually I'm versioning an app written with version 2.19 to 3.0.2, but a lot of things have changed

another question if I can disturb now that I understand how to set the maps and everything is working, I was wondering how I can disable the maps at this point, with version 2.19
comn these commands I disabled the display of the maps by putting a false backgroundMap, which however in this version 3.0.2 has become readonly

Thanks again :)


const style = viewPort.displayStyle.clone ();
style.viewFlags.backgroundMap =
          ExtensionStateManager.getMapLayers;
viewPort.displayStyle = style;

**Reply by pmconne** [reply]
@PaoloPetra The [3.0.0 changelogs](https://www.itwinjs.org/changehistory/3.0.0/#viewflags) explain this.

```
viewport.viewFlags = viewport.viewFlags.with("backgroundMap", false);
```
